### ISSUE TITLE
update on unsigned primary key fails with "Illegal parameters, field id must be a number"

### ISSUE THREAD

**Issue by ochaton** [tarantool/tarantool#6001]
**Bug description**

* OS: MacOS
* Architecture: x86_64

**Affected versions**

```bash
1.10.3-126-ga72c0011a: Tarantool is a rubbish
1.10.9-117-g3aa9b4a4f: Tarantool is a rubbish
2.1.3-9-g85ad8508e: Tarantool is a rubbish
2.2.3-1-g98ecc909a: Tarantool is a rubbish
2.3.3-2-g7eea3a6cb: Tarantool is OK
2.4.3-1-g986fab717: Tarantool is OK
2.5.3-0-gf93e48013: Tarantool is OK
2.6.2-185-gc75656ec2: Tarantool is OK
2.7.1-190-g227af7f4c: Tarantool is OK
2.8.0-0-gefc30ccf8: Tarantool is OK
```

**Steps to reproduce**
```lua
#!/usr/bin/env tarantool
box.cfg{ log_level = 1 }
box.schema.space.create('test', {
	if_not_exists = true,
	format = {
		{ name = 'id',    type = 'unsigned' },
		{ name = 'value', type = 'string'   },
	}
})
box.space.test:create_index('pri', { if_not_exists = true, parts = {'id'} })
box.space.test:replace({ -1ULL, 'first' })

local tuple = assert(box.space.test:pairs():nth(1), "tuple not found? O_o")
if pcall(box.space.test.update, box.space.test, { tuple.id }, {{ '=', 'value', 'second' }}) then
	print(_TARANTOOL..": Tarantool is OK")
	os.exit(0)
else
	print(_TARANTOOL..": Tarantool is a rubbish: "..box.error.last())
	os.exit(1)
end
```

**How to run:**

Just type `tarantool repro.lua`.

**Actual behavior**

Tarantool fails to update tuple which was just selected

**Expected behavior**

This just must work.

**Comment by Totktonada** [comment]
## Interpretation

It is just lack of a feature in 1.10: update by a field name.

## Bisect result

Bisected down to 9ee5cf82ae579d62543a4bb335dd1e99b00dacf6:

```
commit 9ee5cf82ae579d62543a4bb335dd1e99b00dacf6
Author: Vladislav Shpilevoy <v.shpilevoy@tarantool.org>
Date:   Tue May 29 15:33:46 2018 +0300

    tuple: implement update by field name
    
    Tuple fields can be named, accessed by name, indexed by name, but
    till this commit field names couldn't be used in update
    operations. Now it is possible.
    
    This patch is a teaser of updates by JSON path.
    
    Part of #1261
```

## Bisect log

(Added `tostring()` around `box.error.last()`.)

```
2.3.0-0-gce0f2ef8b    bad 
2.3.0-41-gf5d8331e8   bad 
2.3.0-62-g6c627af39   bad 
2.3.0-67-g5a46639f4   bad 
2.3.0-69-gbaa4659c1   bad 
2.3.0-71-g6d4430d0c   bad 
2.3.0-72-g9ee5cf82a   good
2.3.0-83-g165f8ee6f   good
2.3.0-167-g00c6c4377  good
2.3.1-0-g5a1a220ee    good
2.3.2-0-gcb25c5473    good
2.3.3-2-g7eea3a6cb    good
```

**Comment by Totktonada** [comment]
Feel free to reopen if I miss something and didn't spot a real problem.

**Comment by ochaton** [comment]
@Totktonada I suppose that we should backport this fix to 1.10 series

**Comment by Gerold103** [comment]
We backport only bugfixes. This one is not a bug. Field names in update are a feature.
